src/wcs2_0/src/main/java/org/geoserver/wcs2_0/response/WCS20GetCapabilitiesTransformer.java

```diff
@@ -572,7 +582,7 @@ private void handleContents() {
 
             @SuppressWarnings("unchecked")
             final Set<CoverageInfo> coverages =
-                    new TreeSet<CoverageInfo>(new CoverageInfoLabelComparator());
+                    new TreeSet<CoverageInfo>(new CoverageInfoIndexComparator());
```

Comment:
This is not backwards compatible, unless the layer indexes are populated the current output order should not be changed, I have seen existing applications depending on it (as you can see, the previous order was not happening out of a accident, it was forced by a comparator). The compatibility can be re-established by having a delegating or composite comparator, first compare by index, and if the indexes are same, use the other criteria. 
This method may be used to build the chained comparator: https://commons.apache.org/proper/commons-collections/apidocs/org/apache/commons/collections4/ComparatorUtils.html#chainedComparator-java.util.Collection-
(did not check if available in the version of commons we have in the classpath)